Softeq.XToolkit.ProjectInfrastructure/HttpRequest.cs

```diff
@@ -49,6 +53,43 @@ public HttpRequest(Uri uri, HttpMethods method) : this(uri)
             CustomHeaders = new Dictionary<string, object>(headers ?? new Dictionary<string, object>());
         }
 
+        public object Data
+        {
+            get => _data;
+            private set
+            {
+                _serializeDataLock.Wait();
+
+                _data = value;
+                _serializedData = null;
+
+                _serializeDataLock.Release();
+            }
+        }
+
+        public string ContentType { get; private set; }
+
+        public async Task<string> GetSerializedDataAsync()
+        {
+            try
+            {
+                await _serializeDataLock.WaitAsync()
```

Comment:
Acquiring lock should probably be moved outside of the try-finally block. If for some reason WaitAsync() fail - there's no point to release lock (and trying to do so gonna cause another exception, actually)